From 944a8846ba7e2e62b26461f3d7392127376067a1 Mon Sep 17 00:00:00 2001
From: Andiry Xu <jix024@cs.ucsd.edu>
Date: Mon, 5 May 2014 22:37:26 -0700
Subject: [PATCH 7/8] Export vma interval tree operations


Signed-off-by: Andiry Xu <jix024@cs.ucsd.edu>
---
 mm/filemap_xip.c   | 1 +
 mm/interval_tree.c | 6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/mm/filemap_xip.c b/mm/filemap_xip.c
index 5db0594..915f7f6 100644
--- a/mm/filemap_xip.c
+++ b/mm/filemap_xip.c
@@ -177,6 +177,7 @@ void bankshot2_unmap(struct address_space *mapping, struct page *page, unsigned
 		address = vma->vm_start +
 			((pgoff - vma->vm_pgoff) << PAGE_SHIFT);
 		BUG_ON(address < vma->vm_start || address >= vma->vm_end);
+//		printk(KERN_ERR "%s: vma %p, vm_start %lx, vm_end %lx, vm_pgoff %lx, address %lx\n", __func__, vma, vma->vm_start, vma->vm_end, vma->vm_pgoff, address);
 
 		vm_munmap_page(mm, address, PAGE_SIZE);
 
diff --git a/mm/interval_tree.c b/mm/interval_tree.c
index 4a5822a..3fd118a 100644
--- a/mm/interval_tree.c
+++ b/mm/interval_tree.c
@@ -59,6 +59,12 @@ void vma_interval_tree_insert_after(struct vm_area_struct *node,
 			    &vma_interval_tree_augment);
 }
 
+EXPORT_SYMBOL_GPL(vma_interval_tree_insert);
+EXPORT_SYMBOL_GPL(vma_interval_tree_insert_after);
+EXPORT_SYMBOL_GPL(vma_interval_tree_remove);
+EXPORT_SYMBOL_GPL(vma_interval_tree_iter_first);
+EXPORT_SYMBOL_GPL(vma_interval_tree_iter_next);
+
 static inline unsigned long avc_start_pgoff(struct anon_vma_chain *avc)
 {
 	return vma_start_pgoff(avc->vma);
-- 
1.8.1.2

